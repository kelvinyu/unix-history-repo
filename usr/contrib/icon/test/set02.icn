record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("every s1 := insert(set([]),!&lcase) ----> ",image(every s1 := insert(set([]),!&lcase)) | "none")
   write("s2 := set([\"a\",\"aa\",\"ab\",\"b\",1,2,3,4]) ----> ",image(s2 := set(["a","aa","ab","b",1,2,3,4])) | "none")
   write("s3 := s1 ++ s2 ----> ",image(s3 := s1 ++ s2) | "none")
   write("s4 := s1 ** s2 ----> ",image(s4 := s1 ** s2) | "none")
   write("s5 := s1 -- s2 ----> ",image(s5 := s1 -- s2) | "none")
   write("*s3 ----> ",image(*s3) | "none")
   write("every write(!s3) ----> ",image(every write(!s3)) | "none")
   write("*s4 ----> ",image(*s4) | "none")
   write("every write(!s4) ----> ",image(every write(!s4)) | "none")
   write("*s5 ----> ",image(*s5) | "none")
   write("every write(!s5) ----> ",image(every write(!s5)) | "none")
   write("set([]) ++ 'a' ----> ",image(set([]) ++ 'a') | "none")
end
