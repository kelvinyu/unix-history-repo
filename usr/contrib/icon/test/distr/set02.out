every s1 := insert(set([]),!&lcase) ----> none
s2 := set(["a","aa","ab","b",1,2,3,4]) ----> set(8)
s3 := s1 ++ s2 ----> set(32)
s4 := s1 ** s2 ----> set(2)
s5 := s1 -- s2 ----> set(24)
*s3 ----> 32
n
1
o
2
p
3
q
4
r
s
t
u
v
w
x
y
aa
z
ab
a
b
c
d
e
f
g
h
i
j
k
l
m
every write(!s3) ----> none
*s4 ----> 2
a
b
every write(!s4) ----> none
*s5 ----> 24
n
o
p
q
r
s
t
u
v
w
x
y
z
c
d
e
f
g
h
i
j
k
l
m
every write(!s5) ----> none
Run-time error 119 at line 20 in set02.icn
set expected
offending value: 'a'
